app/_components/posts.tsx

```diff
@@ -1,16 +1,47 @@
 import React from 'react';
 
 import { Post } from '@prisma/client';
+import { prisma } from '@/utils/prisma';
 import { SinglePost } from '@/app/_components/singlePost';
+import { REPORTS_TO_HIDE_POST } from '@/utils';
 
-export default async function Posts({ posts }: { posts: Post[] }) {
-  if (posts.length === 0) {
+export interface PostsProps {
+  posts: Post[];
+  showAll?: boolean;
```

Comment:
Ich wuerde nur ungern fuer nur einen Sonderfall (NSFW Seite), extra Logik einbauen, wenn 5/6 Faelle diese Logik nie benutzen.

Stattdessen koennte man by default alle Posts mit zu vielen Reports nicht anzeigen und fuer die NSFW Seite die `SinglePost` Komponente benutzen.

Reply:
Aber dann benutzt man zwei verschiedene Wege das gleiche zu implementieren -> Inkonsistenz
Dann würde ich persönlich lieber mehr Logik in einer Komponente einbauen. Dann muss man Änderungen an der Posts-Komponente nicht zwei mal machen